geniza/corpus/metadata_export.py

```diff
@@ -49,10 +53,30 @@ def get_queryset(self):
         :return: Custom-given query set or query set of all documents
         :rtype: QuerySet
         """
-        qset = self.queryset or self.model.objects.all().metadata_prefetch()
-        qset = qset.prefetch_related(
-            "secondary_languages", "log_entries", "log_entries__user"
-        ).order_by("id")
+        qset = self.queryset or self.model.objects.all()
+        # clear existing prefetches and then add the ones we need,
+        # since admin queryset footnote prefetching conflicts
+        qset = (
+            qset.prefetch_related(None)
+            .metadata_prefetch()
+            .prefetch_related(
+                "secondary_languages",
+                "log_entries",
+                "log_entries__user",
+                Prefetch(
+                    "footnotes",
+                    queryset=Footnote.objects.select_related(
+                        "source",
+                        "source__source_type",
+                    ).prefetch_related(
+                        "source__authorship_set__creator",
+                        "source__languages",
+                        "source__authors",
+                    ),
+                ),
+            )
```

Comment:
Oh wow, tricky. Just curious, do we lose any performance here, or not really since only with the admin interface does this method (of clearing prefetches) do something significantly different?

Reply:
I'm clearing and then reapplying all of the ones we need for this export — there's a discrepancy in how we're prefetching footnotes for admin and footnotes for the data export (now that we need more details for the data export), and this seemed like the simplest solution.